### ISSUE TITLE
Opening context menu on class definition with shared variable name selected signals an error

### ISSUE THREAD

**Issue by Rinzwind** [guillep/pharo-core#16039]
In Pharo 12 build 1258, opening a context menu on a class definition while the name of a shared variable is selected causes a MessageNotUnderstood (“Instance of FallbackMenu did not understand #morphicUIManager”).

In build 1219, doing the same opens the fallback menu (selecting “debug” shows a MessageNotUnderstood was signaled, “Instance of CDSharedVariableNode did not understand #isGlobalClassNameBinding”):
<img width="531" src="https://github.com/pharo-project/pharo/assets/1611248/2d90d398-5dc0-4bc1-9c81-6137e1867bee">

In build 1218, the menu is opened as expected:
<img width="529" src="https://github.com/pharo-project/pharo/assets/1611248/98264a91-d5e0-4be6-bcba-a877c2148905">

**Comment by Ducasse** [comment]
Yes marcus proposed a fix but we lost the AST driven menu - I checked and it was working on P11 on Class name, instance variable and shared var on fluid syntax

**Comment by Ducasse** [comment]
https://github.com/pharo-project/pharo/pull/16035
